packages/operator/pkg/testhelpers/stubclients/stub_clients.go

```diff
@@ -0,0 +1,106 @@
+package stubclients
+
+import (
+	"github.com/odahu/odahu-flow/packages/operator/pkg/apis/training"
+	"github.com/odahu/odahu-flow/packages/operator/pkg/apis/packaging"
+	"github.com/odahu/odahu-flow/packages/operator/pkg/errors"
+	"github.com/odahu/odahu-flow/packages/operator/pkg/repository/util/filter"
+)
+
+type TIStubClient struct {
+	db map[string]training.ToolchainIntegration
+}
+
+func NewTIStubClient() TIStubClient{
+	return TIStubClient{
+		db: make(map[string]training.ToolchainIntegration),
+	}
+}
+
+func (t TIStubClient) GetToolchainIntegration(name string) (*training.ToolchainIntegration, error) {
+	entity, ok := t.db[name]
+	if !ok {
+		return nil, errors.NotFoundError{Entity: name}
+	}
+	return &training.ToolchainIntegration{
+		ID:     name,
+		Spec:   entity.Spec,
+		Status: entity.Status,
+	}, nil
+}
+
+func (t TIStubClient) GetToolchainIntegrationList(
+	options ...filter.ListOption) ([]training.ToolchainIntegration, error) {
+	panic("implement me")
+}
+
+func (t TIStubClient) DeleteToolchainIntegration(name string) error {
+	if _, err := t.GetToolchainIntegration(name); err != nil {
+		return err
+	}
+	delete(t.db, name)
+	return nil
+}
+
+func (t TIStubClient) UpdateToolchainIntegration(md *training.ToolchainIntegration) error {
+	panic("implement me")
+}
+
+func (t TIStubClient) CreateToolchainIntegration(md *training.ToolchainIntegration) error {
+	t.db[md.ID] = training.ToolchainIntegration{
+		ID:    md.ID,
+		Spec:  md.Spec,
+		Status: md.Status,
+	}
```

Comment:
Is it necessary to re-create the `ToolchainIntegration` structure?
```suggestion
	t.db[md.ID] = *md
```